### PR TITLE
Added check for pool

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description

The flow of the automatic fastq delivery is mostly designed around the need for cust001 where pooled rml samples are to be delivered regardless of the quality of individual samples. This however clashes with how we will treat other cases (cust144) where we only want to store (i.e. then transfer) the files ones both the tumor and normal have been sequenced.

### Changed

- The command `cg workflow fastq store-available` now has an extra if clause where it only stores cases that are pools or non-rml cases where all samples have been sequenced and pass QC.

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh fix/apptag-conditional-for-fastq-upload`

### How to test
- [x] do `cg workflow fastq store-available --dry-run`

### Expected test outcome
- [x] check that `hipjackal` is stored while `unitedsloth` is not
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by Vince-janv** [comment]
### Result when one sample in `unitedsloth` has zero reads:
```
[0|0|0] [vincent.janvid@hasta:~] [S_main] 2m57s $ cg workflow fastq store-available --dry-run
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Creating an analysis for case hipjackal
```
### Result after adding reads to second sample in unitedsloth:
```
[0|0|0] [vincent.janvid@hasta:~] [S_main] 20s $ cg workflow fastq store-available --dry-run
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Creating an analysis for case unitedsloth
Creating an analysis for case hipjackal
```

File: cg/cli/workflow/fastq/base.py [hunk 1]
```diff
@@ -48,4 +48,5 @@ def store_available_fastq_analysis(context: click.Context, dry_run: bool = False
     """Creates an analysis object in status-db for all fastq cases to be delivered"""
     status_db: Store = context.obj.status_db
     for case in status_db.cases_to_analyze(pipeline=Pipeline.FASTQ, threshold=False):
-        context.invoke(store_fastq_analysis, case_id=case.internal_id, dry_run=dry_run)
+        if status_db.get_case_pool(case_id=case.internal_id) or case.all_samples_pass_qc:
```
Comment (patrikgrenfeldt) [inline cg/cli/workflow/fastq/base.py:51]:
```suggestion
        if is_pool(case_id=case.internal_id) or case.all_samples_pass_qc:
```

File: cg/store/api/findbusinessdata.py [hunk 2]
```diff
@@ -208,6 +208,38 @@ def families_by_subject_id(
...
+            self.Sample.query.filter(models.Sample.ticket_number == ticket_id)
+            .first()
+            .customer.internal_id
+        )
+
+    def get_samples_from_ticket(self, ticket_id: int) -> List[models.Sample]:
+        return self.query(models.Sample).filter(models.Sample.ticket_number == ticket_id).all()
+
+    def get_samples_from_flowcell(self, flowcell_id: str) -> List[models.Sample]:
+        logging.error("CALLED!!!11!")
```
Comment (patrikgrenfeldt) [inline cg/store/api/findbusinessdata.py:228]:
```suggestion
```

File: cg/store/api/findbusinessdata.py [hunk 3]
```diff
@@ -208,6 +208,38 @@ def families_by_subject_id(
...
+        return (
+            self.Sample.query.filter(models.Sample.ticket_number == ticket_id)
+            .first()
+            .customer.internal_id
+        )
+
+    def get_samples_from_ticket(self, ticket_id: int) -> List[models.Sample]:
+        return self.query(models.Sample).filter(models.Sample.ticket_number == ticket_id).all()
+
+    def get_samples_from_flowcell(self, flowcell_id: str) -> List[models.Sample]:
```
Comment (patrikgrenfeldt) [inline cg/store/api/findbusinessdata.py:227]: Would this make sense?
```suggestion
    def get_samples_from_flowcell(self, flowcell_name: str) -> List[models.Sample]:

```

File: cg/store/api/findbusinessdata.py [hunk 4]
```diff
@@ -417,3 +449,9 @@ def samples_in_customer_group(
             else records
         )
         return records.order_by(models.Sample.created_at.desc())
+
+    def get_case_pool(self, case_id: str) -> models.Pool:
```
Comment (patrikgrenfeldt) [inline cg/store/api/findbusinessdata.py:453]:
```suggestion
    def get_case_pool(self, case_id: str) -> Optional[models.Pool]:

```

File: tests/store/api/add/test_store_add_base.py [hunk 5]
```diff
@@ -79,48 +79,20 @@ def test_add_microbial_sample(base_store: Store, helpers):
...
-    # WHEN adding a new pool into the database
-    new_pool = store.add_pool(
-        customer=new_customer,
-        name="Test",
-        order="Test",
-        ordered=dt.today(),
+    customer = rml_pool_store.customers()[0]
+    application = rml_pool_store.application(tag="RMLP05R800")
+    app_version = rml_pool_store.application_version(application=application, version=1)
+    new_pool = rml_pool_store.add_pool(
```
Comment (patrikgrenfeldt) [inline tests/store/api/add/test_store_add_base.py:87]:
```suggestion
   # WHEN adding a new pool
    new_pool = rml_pool_store.add_pool(
```

File: cg/store/api/findbusinessdata.py [hunk 6]
```diff
@@ -208,6 +208,38 @@ def families_by_subject_id(
...
+    def get_samples_from_ticket(self, ticket_id: int) -> List[models.Sample]:
+        return self.query(models.Sample).filter(models.Sample.ticket_number == ticket_id).all()
+
+    def get_samples_from_flowcell(self, flowcell_id: str) -> List[models.Sample]:
+        logging.error("CALLED!!!11!")
+        flowcell = self.query(models.Flowcell).filter(models.Flowcell.name == flowcell_id).first()
+        if flowcell:
+            return flowcell.samples
+
+    def get_ticket_from_case(self, case_id: str):
```
Comment (patrikgrenfeldt) [inline cg/store/api/findbusinessdata.py:233]: please add return type

File: cg/store/api/findbusinessdata.py [hunk 7]
```diff
@@ -208,6 +208,38 @@ def families_by_subject_id(
...
+
+    def get_customer_id_from_ticket(self, ticket_id: int) -> str:
+        """Returns the customer related to given ticket"""
+        return (
+            self.Sample.query.filter(models.Sample.ticket_number == ticket_id)
+            .first()
+            .customer.internal_id
+        )
+
+    def get_samples_from_ticket(self, ticket_id: int) -> List[models.Sample]:
```
Comment (patrikgrenfeldt) [inline cg/store/api/findbusinessdata.py:224]: please add return type

**Review by patrikgrenfeldt — APPROVED** [review]
👍

**Comment by Vince-janv** [comment]
Deployed in Stage and Prod:
```
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/production/servers/resources/hasta.scilifelab.se
done.
Log deploy... done.
Running --version on each Clinical-Genomics developed tool to verify that all requirements are met:
balsamic 6.0.0 requires colorclass, which is not installed.
argcomplete 1.12.1 has requirement importlib-metadata<3,>=0.23, but you have importlib-metadata 4.11.3.
cg, version 22.60.0
housekeeper, version 3.2.0
demux, version 5.10.3
cglims, version 1.3.4
[0|0|0] θ66° [vincent.janvid@hasta:/home/proj/production/servers/resources/hasta.scilifelab.se] [P_main] master* 2m59s ±
```

**Pull request merged**